cpp/src/arrow/compute/kernels/scalar_temporal_test.cc

```diff
@@ -102,6 +102,44 @@ TEST(ScalarTemporalTest, TestTemporalComponentExtraction) {
   CheckScalarUnary("microsecond", unit, times, int64(), microsecond);
   CheckScalarUnary("nanosecond", unit, times, int64(), nanosecond);
   CheckScalarUnary("subsecond", unit, times, float64(), subsecond);
+
+  auto timestamps = ArrayFromJSON(unit, times);
```

Comment:
Instead of putting all checks in a single test function, can you define a test class that initializes the test vectors and then define separate test "methods" (using `TEST_F`)?

Reply:
Done.

Reply:
Can now add a `TEST_F(ScalarTemporalTest, DayOfWeek)` :-)

Reply:
Done :).